File: MapReduce/src/main/java/com/mine/utils/ConfProperty.java
package com.mine.utils;

import org.apache.hadoop.conf.Configuration;

public class ConfProperty {
  private static final String QUEUE = "queue.name";
  private static final String CLUSTER = "dfs.cluster";
  private static final String NN1 = "cluster.nn1";
  private static final String NN2 = "cluster.nn2";

  private Configuration configuration = new Configuration();

  public ConfProperty() {
  }

  public ConfProperty(String name) {
    init(name);
  }

  private void init(String name) {
    PropertyGet propertyGet = new PropertyGet(name);
    String cluster = propertyGet.get(CLUSTER);
    configuration.set("mapred.job.queue.name", "root." + propertyGet.get(QUEUE));
    configuration.set("fs.defaultFS", "hdfs://" + cluster);
    configuration.set("dfs.nameservices", cluster);
    configuration.set("dfs.ha.namenodes." + cluster, "nn1,nn2");
    configuration.set("dfs.namenode.rpc-address." + cluster + ".nn1", propertyGet.get(NN1));
    configuration.set("dfs.namenode.rpc-address." + cluster + ".nn2", propertyGet.get(NN2));
    configuration.set("dfs.client.failover.proxy.provider.mycluster", "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider");
  }

  public Configuration conf() {
    return configuration;
  }
}

File: MapReduce/src/main/scala/cn/mine/hdfs/ConfProperty.scala
package cn.mine.hdfs

import org.apache.hadoop.conf.Configuration

import java.util.Properties

/**
  * @author ximing.wei
  */
object ConfProperty {
    def apply(): ConfProperty = new ConfProperty()

    def apply(fileName: String): ConfProperty = new ConfProperty(fileName)
}

class ConfProperty {
    private val QUEUE: String       =   "queue.name"
    private val CLUSTER: String     =   "dfs.cluster"
    private val NN1: String         =   "cluster.nn1"
    private val NN2: String         =   "cluster.nn2"

    private val configuration: Configuration = new Configuration

    def this(fileName: String) {
        this()
        init(fileName)
    }

    private def init(name: String): Unit = {
        val properties = new Properties()
        properties.load(this.getClass.getClassLoader.getResourceAsStream(name))
        val cluster = properties.getProperty(CLUSTER)
        configuration.set("mapred.job.queue.name", "root." + properties.getProperty(QUEUE))
        configuration.set("fs.defaultFS", "hdfs://" + cluster)
        configuration.set("dfs.nameservices", cluster)
        configuration.set("dfs.ha.namenodes." + cluster, "nn1,nn2")
        configuration.set("dfs.namenode.rpc-address." + cluster + ".nn1", properties.getProperty(NN1))
        configuration.set("dfs.namenode.rpc-address." + cluster + ".nn2", properties.getProperty(NN2))
        configuration.set("dfs.client.failover.proxy.provider." + cluster, "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider")
    }

    def conf: Configuration = configuration
}
